src/praisonai-agents/praisonaiagents/agent/chat_mixin.py

```
@@ -248,9 +248,48 @@ def _build_system_prompt(self, tools=None):
+        # Apply model-aware runtime profile overrides (opt-in). When no profile
+        # is configured for the active model family, the resolved profile is the
+        # behaviour-neutral "default" whose apply_system_prompt is a no-op, so
+        # the generated prompt is byte-for-byte identical to before.
+        try:
+            profile = self._resolve_runtime_profile()
+            if profile is not None:
+                # apply_system_prompt is a pure no-op unless the profile declares
+                # prompt overrides, so the default/family profiles leave the
+                # generated prompt byte-for-byte identical.
+                system_prompt = profile.apply_system_prompt(system_prompt)
+        except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)